static/js/util.js

```diff
@@ -230,6 +230,20 @@ exports.is_all_or_everyone_mentioned = function (message_content) {
     return all_everyone_re.test(message_content);
 };
 
+exports.move_array_elements_to_front = function util_move_array_elements_to_front(array, selected) {
+    var selected_hash = {};
+    for (var i = 0; i < selected.length; i+=1) {
+        selected_hash[selected[i]] = true;
+    }
+    var selected_elements = array.filter(function (element) {
+        return selected_hash[element];
+    });
+    var unselected_elements = array.filter(function (element) {
+        return !selected_hash[element];
+    });
+    return selected_elements.concat(unselected_elements);
```

Comment:
I'm not javascript expert but maybe you could try to implement this version:
```
1. Create new Array (empty)
2. Iterate over array but from the last index to 0:
   2.1 If element exists in selected as well - add it at the beginning
   2.2 else - add it at the end
```
What do you think?